test/test.js

```diff
@@ -0,0 +1,276 @@
+      });
+
+    expect(res).to.have.status(409);
+  });
+
+  it('should fail when a mandatory field is missing', async () => {
+    const res = await chai.request(karenEndpoint)
+      .post('/karen/v1/users/auth')
+      .send({
+        name: 'Foo Bar',
+        email: 'foo@bar.baz',
+      });
+
+    expect(res).to.have.status(400);
+  });
+
+  it('should fail when sending a malformed request body', async () => {
+    const res = await chai.request(karenEndpoint)
+      .post('/karen/v1/users')
+      .send('this is bad');
+
+    expect(res).to.have.status(400);
+  });
+});
+
+describe('GET /karen/v1/users/self', () => {
+  it('should retrieve the session user', async () => {
+    const res = await chai.request(karenEndpoint)
+      .get('/karen/v1/users/self')
+      .set('User-ID', preExistingUser.id);
+
+    expect(res).to.have.status(200);
+    expect(res.body).to.have.keys('id', 'name', 'email');
+    expect(res.body).to.have.property('name', preExistingUser.name);
+    expect(res.body).to.have.property('email', preExistingUser.email);
+  });
+
+  it('should retrieve the session user with specified fields', async () => {
+    const res = await chai.request(karenEndpoint)
+      .get('/karen/v1/users/self')
+      .query({ includes: ['name', 'email'] })
+      .set('User-ID', preExistingUser.id);
+
+    expect(res).to.have.status(200);
+    expect(res.body).to.have.keys('name', 'email');
+    expect(res.body).to.have.property('name', preExistingUser.name);
+    expect(res.body).to.have.property('email', preExistingUser.email);
+  });
+
+  it('should fail when the user does not exist', async () => {
+    const res = await chai.request(karenEndpoint)
+      .get('/karen/v1/users/self')
+      .set('User-ID', 999999999);
+
+    expect(res).to.have.status(404);
+  });
+
+  it('should fail when sending a bad query', async () => {
+    const res = await chai.request(karenEndpoint)
+      .get('/karen/v1/users/self')
+      .query({ includes: ['name', 'email', 'foo'] })
+      .set('User-ID', preExistingUser.id);
+
+    expect(res).to.have.status(400);
+  });
+});
+
+describe('GET /karen/v1/users/{user_id}', () => {
+  it('should retrieve the specified user', async () => {
+    const res = await chai.request(karenEndpoint)
+      .get(`/karen/v1/users/${preExistingUser.id}`)
+      .set('User-ID', 123);
+
+    expect(res).to.have.status(200);
+    expect(res.body).to.have.keys('id', 'name', 'email');
+    expect(res.body).to.have.property('name', preExistingUser.name);
+    expect(res.body).to.have.property('email', preExistingUser.email);
+  });
+});
+
+describe('PATCH /karen/v1/users/self', () => {
+  it('should update a user', async () => {
+    const res = await chai.request(karenEndpoint)
+      .patch('/karen/v1/users/self')
+      .set('User-ID', preExistingUser.id)
+      .send({
+        name: 'New Name',
+        email: 'newemail@foo.bar',
+        password: 'newpassowrd',
+        avatar_url: 'example.com/newavatar.png',
+      });
+
```

Comment:
Should maybe add a GET call to check that the user was updated properly.

Reply:
Yeah, I guess so. But then we should probably also add a GET call in the POST and DELETE tests that modify the database too, right?

Reply:
Yeah definitely. I think I just forgot haha.